route/build/src/standalone/model_setup.f90

```diff
@@ -127,28 +129,186 @@ SUBROUTINE init_runoff_data(pid,           & ! input: proc id
+  !if(ierr/=0)then; ierr=20; message=trim(message)//'problem allocating space for forcFileInfo'; return; end if
+  print*, "infileinfo is allocated"
+
+  ! poputate the forcingInfo structure with filenames, julian day of sart and end of the simulation
+  do iFile=1,nFile
+
+   ! split the line into "words" (expect one word: the file describing forcing data for that index)
+   read(dataLines(iFile),*,iostat=ierr) filenameData
+   if(ierr/=0)then; message=trim(message)//'problem reading a line of data from file ['//trim(infilename)//']'; return; end if
+
+   ! set forcing file name
+   infileinfo_data(iFile)%infilename = trim(filenameData)
+   print*, infileinfo_data(iFile)%infilename
+
+   ! get the time units
+   call get_var_attr(trim(input_dir)//trim(infileinfo_data(iFile)%infilename), &
+                          trim(vname_time), 'units', infileinfo_data(iFile)%unit, ierr, cmessage)
+   if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+   print*, "unit is", infileinfo_data(iFile)%unit
+
+   ! get the calendar
+   call get_var_attr(trim(input_dir)//trim(infileinfo_data(iFile)%infilename), &
+                          trim(vname_time), 'calendar', infileinfo_data(iFile)%calendar, ierr, cmessage)
+   if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+   print*, "calendar is", infileinfo_data(iFile)%calendar
+
+   ! get the dimension of the time to populate nTime and pass it to the get_nc file
+   call get_nc_dim_len(trim(input_dir)//trim(infileinfo_data(iFile)%infilename), &
+                       trim(dname_time), infileinfo_data(iFile)%nTime, ierr, cmessage)
+   if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+   print*, "time dimension is ", infileinfo_data(iFile)%nTime
+   nTime = infileinfo_data(iFile)%nTime ! the length of time varibale for each nc file
+   print*, "nTime ", nTime
+
+   ! allocate space for time varibale of each file
+   if(allocated(infileinfo_data(iFile)%timeVar)) deallocate(infileinfo_data(iFile)%timeVar)
+   allocate(infileinfo_data(iFile)%timeVar(nTime))
+   !allocated(infileinfo_data(iFile)%timeVar(infileinfo_data(iFile)%nTime))
+   print*, "time var is allocated"
+
+   ! get the time varibale
+   call get_nc(trim(input_dir)//trim(infileinfo_data(iFile)%infilename), &
+               vname_time, infileinfo_data(iFile)%timeVar, 1, nTime, ierr, cmessage) ! does it needs timeVar(:)
+   print*, "time var is", infileinfo_data(iFile)%timeVar
+
+   ! get the time multiplier needed to convert time to units of days for each file
+   select case( trim( infileinfo_data(iFile)%unit(1:index(infileinfo_data(iFile)%unit,' ')) ) )
+    case('seconds'); convTime2Days=86400._dp
+    case('minutes'); convTime2Days=1440._dp
+    case('hours');   convTime2Days=24._dp
+    case('days');    convTime2Days=1._dp
+    case default;    ierr=20; message=trim(message)//'unable to identify time units'; return
```

Comment:
can allow 'second', 'sec' and 's' so can change it to case('seconds','second','sec','s'). Similarly, 
case('minutes','minute','min'),  case('hours','hour','hr','h'), case('days','day','d')
also more meaningful error message like `'<time_units>= '//trim(t_unit)//': <time_units> must be seconds, minutes, hours or days.'` after trim(message)//